src/vs/vscode.proposed.d.ts

```diff
@@ -1781,4 +1781,17 @@ declare module 'vscode' {
 	}
 
 	//#endregion
+
+	//#region https://github.com/microsoft/vscode/issues/90517
+
+	export class FileSystemError2 extends FileSystemError {
+		/**
+		 * A code that identifies this error.
+		 *
+		 * Possible values are names of errors, like [`FileNotFound`](#FileSystemError.FileNotFound), or `'Unknown'`.
+		 */
+		readonly code: string;
+	}
```

Comment:
Things get simpler when you define `FileSystemError` as interface here. That means you don't need to export it `extHost.api.impl` and stll you all type safety

Reply:
Good point. Made it an interface.